packages/region/src/services/region-module.ts

```diff
@@ -223,37 +183,167 @@ export default class RegionModuleService<
     data?: UpdatableRegionFields,
     @MedusaContext() sharedContext: Context = {}
   ): Promise<Region[]> {
-    let result: Region[] = []
+    let normalizedInput: UpdateRegionDTO[] = []
     if (isString(idOrSelectorOrData)) {
-      result = await this.regionService_.update(
-        [{ id: idOrSelectorOrData, ...data }],
-        sharedContext
-      )
+      normalizedInput = [{ id: idOrSelectorOrData, ...data }]
     }
 
     if (Array.isArray(idOrSelectorOrData)) {
-      result = await this.regionService_.update(
-        idOrSelectorOrData,
-        sharedContext
-      )
+      normalizedInput = idOrSelectorOrData
     }
 
     if (isObject(idOrSelectorOrData)) {
-      let toUpdate: Partial<UpdateRegionDTO>[] = []
       const regions = await this.regionService_.list(
-        { ...idOrSelectorOrData },
+        idOrSelectorOrData,
         {},
         sharedContext
       )
 
-      regions.forEach((region) => {
-        toUpdate.push({ id: region.id, ...data })
+      normalizedInput = regions.map((region) => ({
+        id: region.id,
+        ...data,
+      }))
+    }
+    normalizedInput = this.normalizeInput(normalizedInput)
+
+    // If countries are being updated for a region, first make previously set countries' region to null to get to a clean slate.
+    // Somewhat less efficient, but region operations will be very rare, so it is better to go with a simple solution
+    await this.countryService_.update(
+      {
+        selector: {
+          region_id: normalizedInput
+            .filter((region) => !!region.countries)
+            .map((region) => region.id)
+            .flat(),
```

Comment:
todo: extract outside plus add an if, if the filter result in 0 items, then we will have an empty array here

Reply:
Good point, we can avoid a DB request 👍